jvb/src/main/kotlin/org/jitsi/videobridge/Endpoint.kt

```diff
@@ -535,6 +541,75 @@ class Endpoint @JvmOverloads constructor(
         }
     }
 
+    private class AudioSubscriptionEntry {
+        private var excludeWildcard = false
+        private var includeWildcard = true
+        private var wantedSsrcs = mutableListOf<Long>()
```

Comment:
You can use an unmutable List or Set instead, reducing the risk of issue with concurrent modification. 

It looks like you don't need to keep track of `wantedSsrcs` in the case where `includeWildcard || excludeWildcard` is true, because if `includeWildcard` or `excludeWildcard` ever change then `wantedSsrcs` is reset anyway.

Reply:
I just noticed that you're matching the SSRCs against the signaled list. The ReceiverAudioSubscriptionMessage should use source names in its lists, as SSRCs are:
1. Not always known to the receivers
2. May be changed for some receivers with "SSRC rewriting"

So we should:
1. Document the "include" and "exclude" lists' semantics, these are source names (that's why they're strings, not numbers)
2. Change the "getAllRemoteAudioSsrcs" method in Conference to return a list of AudioSourceDesc. I also suggest removing the parameter to this method, and filtering it inside Endpoint.AudioSubscriptionEntry

Reply:
This was my mistake. It now retrieves audioSourceDescs from the Conference and update subscription accordingly.